torch/ao/quantization/_pt2e/prepare.py

```diff
@@ -0,0 +1,123 @@
+import torch
+from .quantizer import Quantizer
+from torch._subclasses import FakeTensor
+from torch.ao.quantization.fx.prepare import (
+    _maybe_insert_input_observers_for_node,
+    _maybe_insert_output_observer_for_node,
+    _is_observer_in_same_graph,
+    _maybe_make_input_output_share_observers,
+    _remove_output_observer,
+    _maybe_insert_observers_before_graph_output,
+    _save_state
+)
+from torch.fx import GraphModule
+from torch.fx import Node
+
+from torch.ao.quantization import QConfigMapping
+from torch.ao.quantization.fx.custom_config import PrepareCustomConfig
+from typing import Dict, Tuple
+
+def _maybe_insert_input_and_output_observers_for_node(
+    node: Node,
+    model: torch.fx.GraphModule
+):
+    this_node_dtype_info = node.meta["target_dtype_info"] if "target_dtype_info" in node.meta else None
+    if "val" in node.meta:
+        output_is_a_tensor = (
+            this_node_dtype_info is not None and
+            isinstance(node.meta["val"], FakeTensor)
+        )
+    else:
+        output_is_a_tensor = this_node_dtype_info is not None
+
+    skip_inserting_observers = (
+        not output_is_a_tensor
+    )
+
+    if skip_inserting_observers:
+        return
+
+    named_modules = dict(model.named_modules(remove_duplicate=False))
+
+    _maybe_insert_input_observers_for_node(
```

Comment:
If observer needs to be inserted for input, does this also do the appropriate graph rewrite?